Sources/TuistGenerator/Extensions/Xcodeproj+Extras.swift

```diff
@@ -28,6 +28,16 @@ extension PBXFileElement {
 }
 
 extension PBXBuildFile {
+    /// Apply platform filters either `platformFilter` or `platformFilters` depending on count
+    public func applyPlatformFilters(_ filters: PlatformFilters, applicableTo target: Target) {
+        // If the dependency fewer platforms, apply filters.
+        let dependingTargetPlatformFilters = target.dependencyPlatformFilters
+        if filters.isSubset(of: dependingTargetPlatformFilters) {
```

Comment:
Worth thinking through a few examples:

**A) Consuming target has fewer supported platforms**

`App` > `FrameworkA`

- `App`
  - iOS
- `FrameworkA`
  - iOS
  - visionOS


**B) Consuming target has the same supported platforms**

- `App`
  - iOS
  - visionOS
- `FrameworkA`
  - iOS
  - visionOS

**C) Consuming target has the more supported platforms**

- `App`
  - iOS
  - visionOS
- `FrameworkA`
  - iOS

**D) Consuming target has the different supported platforms**

- `App`
  - iOS
  - macOS
- `FrameworkA`
  - iOS
  - visionOS
 

I think the logic may need to slightly different to accommodate for the last case